shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/GpuHashJoin.scala

```diff
@@ -197,19 +198,21 @@ trait GpuHashJoin extends GpuExec with HashJoin {
       override def hasNext: Boolean = {
         while (nextCb.isEmpty && (first || stream.hasNext)) {
           if (stream.hasNext) {
+            val startTime = System.nanoTime()
```

Comment:
Note that `stream.hasNext` can also be expensive, per the analysis in #658.  I think we need to start timing before the `hasNext` call or the time spent waiting for the streamed table could be similarly underreported.

Reply:
I will see it